lab_2_tokenize_by_bpe/main.py

```diff
@@ -26,31 +38,57 @@ def collect_frequencies(
     :param end_of_word: a token that signifies the end of word
     :return: dictionary in the form of <preprocessed word: number of occurrences>
     """
+    if not(isinstance(text, str) and isinstance(end_of_word, str)) or not(
+        isinstance(start_of_word, str) or start_of_word is None
+    ):
+        return None
+    freq_dict = {}
+    words = text.split()
+    for word in words:
+        preprocessed_word = prepare_word(word, start_of_word, end_of_word)
+        if not preprocessed_word:
+            return None
+        freq_dict.update({preprocessed_word: words.count(word)})
```

Comment:
What if the word is already in the dictionary? Could we save some resources on not counting its occurrences again?

Reply:
I'll work on it

Reply:
While you're working on it, consider also adding a key to the dictionary not via `update`. Btw, why do you think `update` is not very appropriate here?

Reply:
because update can change some of values I have? I mean that update adds elements to the dict if the key is not in the it, otherwise, it updates the key with the new value, and also here we need just to add a new key with its value?